packages/service-library/src/servicelib/application_setup.py

```diff
@@ -0,0 +1,154 @@
+import functools
+import inspect
+import logging
+from enum import Enum
+from typing import Dict, List, Optional
+
+from aiohttp import web
+
+from .application_keys import APP_CONFIG_KEY
+
+log = logging.getLogger(__name__)
+
+APP_SETUP_KEY = f"{__name__ }.setup"
+
+class ModuleCategory(Enum):
+    SYSTEM = 0
+    ADDON = 1
+
+class AppSetupBaseError(Exception):
+    pass
+
+class DependencyError(AppSetupBaseError):
+    pass
+
+def mark_as_module_setup(module_name: str, category: ModuleCategory,*,
```

Comment:
Suggesiton/thought: I would name it simply "module_setup" as "mark_as" comes implicit with the concept of decorator/annotation

Reply:
ok. renamed as ``app_module_setup`` to  distinguish from python modules